### PR TITLE
RESTWS-829: Add REST resource for Program Management

### PR DESCRIPTION
**Pull request by BartlomiejRasztabiga**
## Description of what I changed
I've added creatableProperties for two resources to allow users to create them through REST API.
I've also added the ability to delete ProgramWorkflowState since it's already supported in Legacy UI.

PR with documentation: https://github.com/openmrs/openmrs-contrib-rest-api-docs/pull/157

## Issue I worked on
see https://issues.openmrs.org/browse/RESTWS-829

## Checklist: I completed these to help reviewers :)
- [X] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Do you plan to add some tests for these new changes?

**Comment by BartlomiejRasztabiga** [comment]
> Do you plan to add some tests for these new changes?

Yes, I'm adding some right now but I've encountered errors with Jackson:
`org.codehaus.jackson.map.JsonMappingException: Conflicting getter definitions for property "set": org.openmrs.Concept#isSet(0 params) vs org.openmrs.Concept#getSet(0 params)`
It's because Concept has two getters for the same param and Jackson gets confused.

**Comment by BartlomiejRasztabiga** [comment]
@dkayiwa Here's the exception if you're curious: https://github.com/openmrs/openmrs-module-webservices.rest/pull/492/commits/7f25074f9f59d1834004fb7d87964502f199fe0c

**Comment by BartlomiejRasztabiga** [comment]
Nevermind, I've fixed that by moving the test to omod-2.0 module :)

**Comment by BartlomiejRasztabiga** [comment]
I've just created test for creating ProgramWorkflow: https://github.com/openmrs/openmrs-module-webservices.rest/pull/492/commits/c20db5ff7f5abbd8e59ab42dca38c6a945880347

**Comment by dkayiwa** [comment]
@BartlomiejRasztabiga i knew you would fix it by yourself. 😊

**Comment by BartlomiejRasztabiga** [comment]
@dkayiwa I've added tests for these two resources I've modified.

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ProgramWorkflowController2_0Test.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.Test;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class ProgramWorkflowController2_0Test extends MainResourceControllerTest {
+
+	@Override
+	@Test(expected = Exception.class)
```
Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ProgramWorkflowController2_0Test.java:25]: Do we expect the generic exception or a more specific exception class?

Comment (BartlomiejRasztabiga) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ProgramWorkflowController2_0Test.java]: https://github.com/openmrs/openmrs-module-webservices.rest/pull/492/commits/32d0eba524e346003aac650f03f7ba902d7d5040

**Pull request merged**